Tests/SwiftDocCTests/Infrastructure/ResolvedTopicReferenceTests.swift

```diff
@@ -78,4 +78,33 @@ class ResolvedTopicReferenceTests: XCTestCase {
             _ = topicReference.absoluteString
         }
     }
+    
+    func testResolvedTopicReferenceDoesNotLeakMemory() throws {
+        #if os(Linux) || os(Android)
+            throw XCTSkip("'autoreleasepool' is not supported on this platform so this test can not be run reliably.")
+        #endif
+        
+        autoreleasepool {
+            var topicReference: ResolvedTopicReference? = ResolvedTopicReference(
+                bundleIdentifier: "com.apple.example",
+                path: "/documentation/path/sub-path",
+                fragment: nil,
+                sourceLanguage: .swift
+            )
+            XCTAssertNotNil(topicReference)
+            
+            ResolvedTopicReference.sharedPool.sync { pool in
+                XCTAssertNotNil(pool.first?.value.values.first?.value)
+                XCTAssertFalse(pool.isEmpty)
+            }
+            
+            topicReference = nil
+            XCTAssertNil(topicReference)
```

Comment:
I believe this will always hold true, regardless of the changes introduced by this PR. What are we actually trying to test here? That the underlying storage is no longer being referenced?

Reply:
Ditto

Reply:
This is just there to add a use of `topicReference` so that Swift doesn't throw a warning or potentially optimize this all out. I'll add a comment to clarify that.